### ISSUE TITLE
Swagger Documentation Not Reflecting Array Objects in PSU API

### ISSUE THREAD

**Issue by RubenT91** [ironmansoftware/powershell-universal#3874]
### Version

5.0.9

### Severity

Medium

### Environment

msi

### Steps to Reproduce

Create an API endpoint in PSU that returns an array of objects. For example:
New-PSUEndpoint -Url "/distributionlists" -Method GET -Endpoint {
    $distributionLists = @()
    
    $distributionLists += [PSCustomObject]@{
        DisplayName = "Example Name"
        SamAccountName = "exampleSAM"
        PrimarySmtpAddress = "example@domain.com"
        DistinguishedName = "CN=example,OU=exampleOU,DC=domain,DC=com"
        ManagedBy = @("Manager1", "Manager2")
        Backup = @()
        RequireSenderAuthenticationEnabled = $true
        proxyAddresses = @("SMTP:example@domain.com")
        msExchBypassModerationLink = @()
        msExchModeratedByLink = @()
        msExchCoManagedByLink = @("Manager1")
        ShowInAddressbook = @("AddressBook1", "AddressBook2")
    }
    
    $distributionLists | ConvertTo-Json
}

Configure the PSU Swagger documentation via the Endpoint Documentation Feature in the PSU admin console.

Verify the API response is correctly returning an array of objects when queried.

Navigate to /swagger/index.html in the PSU admin console to view the automatically generated Swagger documentation for this API endpoint.

Observe that while the API response is correct (returning an array), Swagger incorrectly shows the return type as a single object instead of an array of objects.

### Expected behavior

```console
Swagger should correctly display the response type for the API as an array of objects in the documentation. The schema should show type: array with the appropriate object definition inside it.
```

### Actual behavior

```console
Despite the API returning a valid JSON array, Swagger documentation does not reflect this correctly. Instead, it displays the response as a single object rather than an array of objects. Custom schema adjustments through appsettings.json did not resolve the issue either.
```

### Additional Environment data

PSU Version: 5.0.8 (or customer-specific version)
Operating System: Windows Server 2022
Dashboard Framework: Not applicable
Other: Custom schemas applied through appsettings.json did not reflect properly after service restart. PSU logs do not show any errors related to Swagger documentation generation.

**Comment by adamdriscoll** [comment]
I have verified that this is working in 5.0.10. This is the output. 

![image](https://github.com/user-attachments/assets/ff9a4182-4692-4ea1-819b-b51c8512d0d9)

This is how it was configured. 

endpoint doc implementation: 

```

    [Documentation()]
    class CustomObjectType {
        [string[]]$ShowInAddressbook
    }

```

Endpoint definition. 

```
<#
.OUTPUTS
200:
  Description: This is an output value. 
  Content:
      application/json: CustomObjectType

400:
  Description: Invalid input
#>
param()

$distributionLists += [PSCustomObject]@{
    DisplayName = "Example Name"
    SamAccountName = "exampleSAM"
    PrimarySmtpAddress = "example@domain.com"
    DistinguishedName = "CN=example,OU=exampleOU,DC=domain,DC=com"
    ManagedBy = @("Manager1", "Manager2")
    Backup = @()
    RequireSenderAuthenticationEnabled = $true
    proxyAddresses = @("SMTP:example@domain.com")
    msExchBypassModerationLink = @()
    msExchModeratedByLink = @()
    msExchCoManagedByLink = @("Manager1")
    ShowInAddressbook = @("AddressBook1", "AddressBook2")
}

$distributionLists | ConvertTo-Json
```

**Comment by adamdriscoll** [comment]
User is asking about the entire object return being an array and not a property of the object.

**Comment by adamdriscoll** [comment]
There is smaller fix we need to include in 5.0.11 to support custom types for this. Basic types, like strings, will work as is. To specify an array is to be returned, use this syntax. 

```
<#
.OUTPUTS
200:
  Description: This is an output value. 
  Content:
      application/json: CustomObjectType[]  

400:
  Description: Invalid input
#>
param()
```
